hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/DataSerializableSerializer.java

```diff
@@ -190,19 +190,26 @@ private IOException rethrowReadException(int id, int factoryId, String className
                 + ", exception: " + e.getMessage(), e);
     }
 
-    private Exception tryClarifyInstantiationException(Class aClass, Exception exception) {
-        if (!(exception instanceof InstantiationException)) {
+    /**
+     * @return
+     *         <ul>
+     *         <li>If {@code exception} is an {@link NoSuchMethodError} and matches criteria of
+     *         {@link #tryGenerateClarifiedExceptionMessage(Class)}, a new {@link InstantiationException} with the new message
+     *         <li>Otherwise, {code exception}
+     *         </ul>
+     */
+    private Exception tryClarifyInstantiationException(Class<?> aClass, Exception exception) {
```

Comment:
Nit-pick: since this now handles `ReflectiveOperationException` instead of `InstantiationException`, perhaps we should update the method name?

Reply:
0c14859f0683e8a7cc91cdc8f5d38ad63b900c5b